src/Mapper/Type/ObjectType.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace MyTarget\Mapper\Type;
+
+use Doctrine\Common\Annotations\Reader;
+use Doctrine\Instantiator\Instantiator;
+use Doctrine\Instantiator\InstantiatorInterface;
+use MyTarget\Mapper\Annotation\Field;
+use MyTarget\Mapper\Mapper;
+use MyTarget\Mapper\Exception\ClassNotFoundException;
+
+class ObjectType implements Type
+{
+    /**
+     * @var InstantiatorInterface
+     */
+    private $instantiator;
+
+    /**
+     * @var Reader
+     */
+    private $annotations;
+
+    public function __construct(Reader $annotations, InstantiatorInterface $instantiator = null)
+    {
+        $this->annotations = $annotations;
+        $this->instantiator = $instantiator ?: new Instantiator();
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function hydrated($value, $type, Mapper $mapper)
+    {
+        if ( ! class_exists($type)) {
+            throw new ClassNotFoundException($type);
+        }
+
+        $class = new \ReflectionClass($type);
+        $result = $this->instantiator->instantiate($type);
+
+        do {
+            foreach ($class->getProperties() as $property) {
```

Comment:
If a parent class has a property that is defined as protected or public it is processed many times. Perhaps documentation should specify access level of annotated properties. To make it more evident one could write: `$class->getProperties(\ReflectionProperty::IS_PRIVATE)`